### PR TITLE
Reconnect desktop

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers @maxtaco 
This allows the desktop to reconnect to the socketfile. Not ideal but I wanted to not touch framed-msgpack-rpc at all

### CODE REVIEW

File: react-native/react/engine/transport.desktop.js [hunk 1]
```diff
@@ -14,17 +25,12 @@ export default class DesktopTransport extends BaseTransport {
       console.error('No keybased socket file found!')
     }
 
-    let hooks = null
-    if (connectCallback) {
-      hooks = {connected: connectCallback}
-    }
+    this.net_opts.path = sockfile
```
Comment (chrisnojima) [inline react-native/react/engine/transport.desktop.js]: Kinda sucks

File: react-native/react/engine/transport.desktop.js [hunk 2]
```diff
@@ -4,6 +4,17 @@ import {socketPath} from '../constants/platform.native.desktop'
...
+    if (connectCallback) {
+      hooks = {connected: connectCallback}
+    }
+
+    super({hooks}, null, incomingRPCCallback)
+    this.needsConnect = true
+    this.needsBase64 = false
+  }
+
+  _connect_critical_section (cb) {
```
Comment (chrisnojima) [inline react-native/react/engine/transport.desktop.js]: also kinda sucks

File: react-native/react/engine/transport.desktop.js (hunk 1 above)
Comment (MarcoPolo) [inline react-native/react/engine/transport.desktop.js]: don't you mean it kinda _socks_... 


I'll see myself out.

File: react-native/react/engine/transport.desktop.js [hunk 3]
```diff
@@ -14,17 +25,12 @@ export default class DesktopTransport extends BaseTransport {
...
-      {path: sockfile, hooks},
-      null,
-      incomingRPCCallback
-    )
-    this.needsConnect = true
-    this.needsBase64 = false
+    if (sockfile) {
+      super._connect_critical_section(cb)
+    } else {
+      cb(new Error('No socketfile'))
```
Comment (MarcoPolo) [inline react-native/react/engine/transport.desktop.js]: so whoever calls this knows to retry when it gets an error in the callback?

where does the retry happen?

Comment (chrisnojima) [inline react-native/react/engine/transport.desktop.js]: This is all driven by framed msg pack

**Comment by MarcoPolo** [comment]
:+1:

**Comment by chrisnojima** [comment]
https://github.com/keybase/client/pull/1385 is way better than this

**Pull request closed without merging**